torch/_inductor/codegen/triton.py

```diff
@@ -1050,45 +1050,25 @@ def mask_loads(self, mask):
         finally:
             self._load_mask = prior
 
-    def gen_assert_indirect_indexing(self, buffer, original_index, mask):
-        if mask == "None":
-            return
-        body = self.current_node._body
-        indirect_size = dict(zip(body.indirect_vars, body.indirect_max_sizes))
-        indirect_name = body.indirect_new
-        # Many indirect variables may be mapped to the same CSE'd variable
-        # For example when you do x[y, y] for x = randn(3, 8)
-        var_size = collections.defaultdict(set)
-        for ind, size in indirect_size.items():
-            var_size[indirect_name[ind]].add(V.kernel.rename_indexing(size))
-
-        indirect_vars = [
-            s for s in original_index.free_symbols if s.name.startswith("tmp")
-        ]
-        for var in indirect_vars:
-            sizes = list(var_size[var])
-            if all(isinstance(s, sympy.Integer) for s in sizes):
-                size = min(sizes)
-            else:
-                # Should this go here or in TritonPrinter?
-                def print_min(expr):
-                    if len(expr) == 1:
-                        return texpr(expr[0])
-                    else:
-                        return f"min({texpr(expr[0])}, {print_min(expr[1:])})"
-
-                size = print_min(sizes)
+    def indirect_indexing(self, var, size):
+        if config.triton.assert_indirect_indexing and torch.version.hip is None:
             # The conditions need to be in parens because of Python's operator precedence.
             # It'd be less # error-prone to use and/or/not, which is suported by triton
-            cond = f"((0 <= {var}) & ({var} < {size}))"
+            cond = f"(0 <= {var}) & ({var} < {size})"
             cond_print = f"0 <= {var} < {size}"
-            if not isinstance(original_index, sympy.Integer):
-                var_mask = f"({mask})" if "&" in mask else mask
-                var_mask = f" | ~{var_mask}"
-            else:
-                var_mask = ""
-            line = f'tl.device_assert(({cond}){var_mask}, "index out of bounds: {cond_print}")'
-            self.cse.generate(buffer, line, assignment=False)
+
+            mask_vars = list(var.mask_vars)
+            if mask_vars:
+                mask = (
+                    f"{mask_vars[0]}"
+                    if len(mask_vars) == 1
+                    else f"({' & '.join(mask_vars)})"
+                )
+                cond = f"({cond}) | ~{mask}"
+            line = f'tl.device_assert({cond}, "index out of bounds: {cond_print}")'
+            self.cse.generate(self.compute, line, assignment=False)
```

Comment:
Is it alright to always generate this in `self.compute`? Can't the load be in `self.load`, which happens before `self.compute`?

Reply:
Indirect loads are always in the `compute` buffer:
https://github.com/pytorch/pytorch/blob/660a0d8622d79fdc1a2b9a7102a5ecd33096c2ec/torch/_inductor/codegen/common.py#L717-L719